srv/src/main/java/my/bookshop/config/DestinationConfiguration.java

```diff
@@ -20,13 +21,16 @@ public class DestinationConfiguration {
 	void applicationReady(ApplicationReadyEvent ready) {
 		Integer port = environment.getProperty("local.server.port", Integer.class);
 		String destinationName = environment.getProperty("cds.remote.services.'[API_BUSINESS_PARTNER]'.destination.name");
+		Boolean onPremise = environment.getProperty("on.premise", Boolean.class);
 		if(port != null && destinationName != null) {
 			DefaultHttpDestination httpDestination = DefaultHttpDestination
 			.builder("http://localhost:" + port)
 			.name(destinationName).build();
 
 			DestinationAccessor.prependDestinationLoader(
 				new DefaultDestinationLoader().registerDestination(httpDestination));
+		} else if (destinationName == "s4-destination" && onPremise) {
+			DestinationAccessor.getDestination("s4-destination").asHttp().decorate(DefaultErpHttpDestination::new);
```

Comment:
I think this doesn't work. There is currently a workaround required for on-premise destinations, which is described here: https://github.wdf.sap.corp/MA/cap-cloud-sdk-collaboration/issues/55#issuecomment-2948646

Reply:
As discused, the code doesn't work.  The header field must be configured in the destination.